tests/test-data/pkg/expected-warnings-darwin.txt

```diff
@@ -24,6 +24,42 @@
   The file must be distributed with executable as %2.
   %1: node_modules/open/xdg-open
   %2: path-to-executable/xdg-open
+> Warning Cannot include addon %1 into executable.
+  The addon must be distributed with executable as %2.
+  %1: node_modules/@ronomon/direct-io/binding.node
+  %2: path-to-executable/binding.node
+> Warning Cannot include file %1 into executable.
+  The file must be distributed with executable as %2.
+  %1: node_modules/etcher-sdk/node_modules/drivelist/build/Release/drivelist.node
```

Comment:
Because of these changes to expected warnings, I decided to test the draft standalone zip package downloaded from here:
https://github.com/balena-io/balena-cli/releases/tag/untagged-29cb5354571d6faf986b

Which actually still bundles Node.js v10:

```
$ ./balena-cli-12.41.0-draft/balena version -a
balena-cli version "12.41.0"
Node.js version "10.21.0"
```

I tried the `local flash` command (because it exercises the Etcher SDK) and it resulted in an error:
```
$ sudo DEBUG=1 ./balena-cli-12.41.0-draft/balena local flash balena-cloud-test-rpi-raspberrypi3-2.69.1+rev1-dev-v12.3.5.img.zip
Password:
[debug] original argv0="./balena-cli-12.41.0-draft/balena" argv=[/Users/paulo/Downloads/balena-cli-12.41.0-draft/balena,/snapshot/volume/bin/balena,local,flash,balena-cloud-test-rpi-raspberrypi3-2.69.1+rev1-dev-v12.3.5.img.zip] length=5
[debug] new argv=[/Users/paulo/Downloads/balena-cli-12.41.0-draft/balena,/snapshot/volume/bin/balena,local:flash,balena-cloud-test-rpi-raspberrypi3-2.69.1+rev1-dev-v12.3.5.img.zip] length=4
? Select a drive /dev/disk2 (4.1 GB) - TOSHIBA USB Flash Memory Media
? This will erase the selected drive. Are you sure? Yes
EBADF: bad file descriptor, write
Segmentation fault: 11
```

Same error on Linux and macOS. The error does not happen with the master branch, standalone zip package version 12.40.3, so it's something introduced by this PR.

I also tested building the standalone zip package on my laptop using Node.js v12 and Node.js v14, and got the same `EBADF: bad file descriptor, write` error (as above with Node.js v10) -

```
$ ./build-bin/balena version -a
balena-cli version "12.40.3"
Node.js version "12.18.1"

$ sudo DEBUG=1 ./build-bin/balena local flash ~/Downloads/balena-cloud-test-rpi-raspberrypi3-2.69.1+rev1-dev-v12.3.5.img.zip
Password:
[debug] original argv0="./build-bin/balena" argv=[/Users/paulo/balena/balena-io/balena-cli/build-bin/balena,/snapshot/balena-cli/bin/balena,local,flash,/Users/paulo/Downloads/balena-cloud-test-rpi-raspberrypi3-2.69.1+rev1-dev-v12.3.5.img.zip] length=5
[debug] new argv=[/Users/paulo/balena/balena-io/balena-cli/build-bin/balena,/snapshot/balena-cli/bin/balena,local:flash,/Users/paulo/Downloads/balena-cloud-test-rpi-raspberrypi3-2.69.1+rev1-dev-v12.3.5.img.zip] length=4
? Select a drive /dev/disk2 (4.1 GB) - TOSHIBA USB Flash Memory Media
? This will erase the selected drive. Are you sure? Yes
EBADF: bad file descriptor, write

$ nvm use 14
Now using node v14.16.0 (npm v6.14.11)

$ rm -rf node_modules && npm i && npm run build:fast && npm run build:standalone

$ ./build-bin/balena version -a
balena-cli version "12.40.3"
Node.js version "14.4.0"

$ sudo DEBUG=1 ./build-bin/balena local flash ~/Downloads/balena-cloud-test-rpi-raspberrypi3-2.69.1+rev1-dev-v12.3.5.img.zip
[debug] original argv0="./build-bin/balena" argv=[/Users/paulo/balena/balena-io/balena-cli/build-bin/balena,/snapshot/balena-cli/bin/balena,local,flash,/Users/paulo/Downloads/balena-cloud-test-rpi-raspberrypi3-2.69.1+rev1-dev-v12.3.5.img.zip] length=5
[debug] new argv=[/Users/paulo/balena/balena-io/balena-cli/build-bin/balena,/snapshot/balena-cli/bin/balena,local:flash,/Users/paulo/Downloads/balena-cloud-test-rpi-raspberrypi3-2.69.1+rev1-dev-v12.3.5.img.zip] length=4
? Select a drive /dev/disk2 (4.1 GB) - TOSHIBA USB Flash Memory Media
? This will erase the selected drive. Are you sure? Yes
EBADF: bad file descriptor, write
```

So it's not related to the version of Node.js.

I have noticed that the CLI's `node_modules` folder now includes 2 installations of `drivelist`:

```
$ find node_modules -type d -name drivelist
node_modules/drivelist
node_modules/etcher-sdk/node_modules/drivelist
```

The `npm-shrinkwrap.json` file clarifies it's because `etcher-image-write` and `resin-device-operations` require `drivelist` v8:

```
    "etcher-image-write": {
      "version": "9.1.6",
      "resolved": "https://registry.npmjs.org/etcher-image-write/-/etcher-image-write-9.1.6.tgz",
      "requires": {
        "drivelist": "^8.0.10",
        ...
```
```
    "resin-device-operations": {
      "version": "1.7.0",
      "resolved": "https://registry.npmjs.org/resin-device-operations/-/resin-device-operations-1.7.0.tgz",
      "requires": {
        "drivelist": "^8.0.10",
        ...
```
```
    "drivelist": {
      "version": "8.0.10",
      "resolved": "https://registry.npmjs.org/drivelist/-/drivelist-8.0.10.tgz",
```

... while `etcher-sdk` requires `drivelist` v9:

```
    "etcher-sdk": {
      "version": "6.2.0",
      "resolved": "https://registry.npmjs.org/etcher-sdk/-/etcher-sdk-6.2.0.tgz",
      "requires": {
        "drivelist": "^9.2.4",
        ...
      },
      "dependencies": {
        ...,
        "drivelist": {
          "version": "9.2.4",
          "resolved": "https://registry.npmjs.org/drivelist/-/drivelist-9.2.4.tgz",
```

This may or may not have something to do with the `EBADF: bad file descriptor, write` error.

If `etcher-image-write` and `resin-device-operations` could be updated to use `drivelist` v9 like `etcher-sdk`, then we would not need to installations of it.

I thought it might be case of having to add some files to the `pkg.scripts` or `pkg.assets` sections of the CLI's `package.json`:
https://github.com/balena-io/balena-cli/blob/v12.40.3/package.json#L26-L46

But then I realised that I also get the error with the npm installation - a straight `./bin/balena` execution:

```
$ sudo ./bin/balena local flash ~/Downloads/balena-cloud-test-rpi-raspberrypi3-2.69.1+rev1-dev-v12.3.5.img.zip
? Select a drive /dev/disk2 (4.1 GB) - TOSHIBA USB Flash Memory Media
? This will erase the selected drive. Are you sure? Yes
EBADF: bad file descriptor, write
```

So it may actually not have anything to do with the standalone zip package. If it's not the Node.js version and not the standalone zip package, then it may actually be a source code bug somewhere...

Reply:
@pdcastro it just needed 3 extra arguments https://github.com/balena-io-modules/etcher-sdk/blob/cac65b6864619a349000209cf284a9bb110d3068/examples/multi-destination.ts#L119-L121 .